Lib/_weakrefset.py

```diff
@@ -90,7 +90,13 @@ def clear(self):
         self.data.clear()
 
     def copy(self):
-        return self.__class__(self)
+        new = WeakSet()
+        add = new.add
+        for key in self.data.copy():
```

Comment:
This relies on `dict.copy` atomicity which was achieved with BPO-31179 but AFAICT the feature isn't specified explicitly anywhere so it should rather be treated as an implementation detail as it might have regressed since.

Reply:
Do you mean for me to add documentation for atomic dict.copy?

Reply:
Pretty sure he wants you to not have code that relies on dict.copy being atomic, because its atomicity is not guaranteed.